backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -196,20 +232,38 @@ private boolean createPolicyDefinitionForPartner(Partner partner) {
      */
     private boolean createFrameWorkPolicy() {
         var body = edcRequestBodyBuilder.buildFrameworkAgreementPolicy();
-        try {
-            var response = sendPostRequest(body, List.of("v2", "policydefinitions"));
-            boolean result = response.isSuccessful();
-            if (!result) {
-                log.warn("Framework Policy Registration failed \n" + response.body().string());
+        try (var response = sendPostRequest(body, List.of("v2", "policydefinitions"))) {
+            if (!response.isSuccessful()) {
+                log.warn("Framework Policy Registration failed");
+                if (response.body() != null) {
+                    log.warn("Response: \n" + response.body().string());
+                }
+                return false;
             }
-            response.body().close();
-            return result;
+            return true;
         } catch (Exception e) {
             log.error("Failed to register Framework Policy", e);
             return false;
         }
     }
 
+
+    private boolean registerDtrAsset() {
+        var body = edcRequestBodyBuilder.buildDtrRegistrationBody();
+        try (var response = sendPostRequest(body, List.of("v3", "assets"))) {
```

Comment:
Was this on purpose? Is there a v3 asset api already? If so, just ignore this comment.